src/App.tsx

```diff
@@ -15,36 +17,105 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alpha = 'alpha',
+  Length = 'length',
+  Default = 'default',
+}
+
+type SortingFunction = (a: string, b: string) => number;
+
+const SORTING_FUNCTIONS: {
+  [key in Exclude<SortType, SortType.Default>]: SortingFunction;
+} = {
+  [SortType.Alpha]: (a, b) => a.localeCompare(b),
+  [SortType.Length]: (a, b) => a.length - b.length,
+};
+
 export const App: React.FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState<string[]>(goodsFromServer);
+  const [sortField, setSortField] = useState<SortType>(SortType.Default);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
```

Comment:
No need to explicitly set types that can be implied from values; 
Typescript internally handles types; simply use `useState(valueWithImplicitType)`